### DISCUSSION TITLE
How do I check that the key is legacy

### DISCUSSION THREAD

**Discussion by beldmit** [openssl/openssl#21538] [Q&A] [answered]
For non-legacy keys it's safe to make _todata/_fromdata conversions. For engine-backed keys, it still can be done, but the resulting key will not be operable. How do I distinguish that the key is a legacy one (at least for RSA/EC)?

**Comment by mattcaswell** [comment, marked as answer]
`EVP_PKEY_get0_provider()` will return a reference to a provider associated with the key or NULL if there is none (implying it is blank or legacy).

**Comment by beldmit** [comment]
OK. And what about the legacy structs - RSA/EC?

**Reply by mattcaswell** [reply]
Not sure I understand the question. What about them?

**Reply by beldmit** [reply]
I have a legacy application that creates RSA/EC structs. I want to check if a particular struct is suitable to creating a EVP_PKEY via legacy accessors and `EVP_PKEY_fromdata` or not
